File: aoj/volume100/ocaml/A10023.ml
let shuffle cards h =
    (String.sub cards h ((String.length cards)-h)) ^ (String.sub cards 0 h)

let () =
    let rec loop () =
        let cards = ref (read_line ()) in
        match !cards with
        | "-" -> ()
        | _   -> (
            let n_shuffle = read_int () in
            for i = 1 to n_shuffle do
                let h = read_int () in
                cards := shuffle !cards h
            done ;
            Printf.printf "%s\n" !cards ;
            loop ()
        )
    in
    loop ()

File: aoj/volume100/scala/A10023.scala
object Main {
  def main(args: Array[String]) {
    loop
  }

  def loop {
    var cards = scala.io.StdIn.readLine
    if (cards == "-") {
      return
    }

    val n_shuffle = scala.io.StdIn.readInt
    for (_ <- 1 to n_shuffle) {
      val h = scala.io.StdIn.readInt
      cards = cards.slice(h, cards.length) + cards.slice(0, h)
    }
    println(cards)
    loop
  }
}
